vizro-core/tests/helpers/common.py

```diff
@@ -0,0 +1,48 @@
+import subprocess
+
+import cv2
+import imutils
+from hamcrest import assert_that, equal_to
+
+
+def comparison_logic(original_image, new_image):
+    """Comparison process."""
+    difference = cv2.subtract(original_image, new_image)
+    blue, green, red = cv2.split(difference)
+    assert_that(cv2.countNonZero(blue), equal_to(0), reason="Blue channel is different")
+    assert_that(cv2.countNonZero(green), equal_to(0), reason="Green channel is different")
+    assert_that(cv2.countNonZero(red), equal_to(0), reason="Red channel is different")
+
+
+def create_image_difference(original, new):
+    """Creates new image with diff of images comparison."""
+    diff = original.copy()
+    cv2.absdiff(original, new, diff)
+    gray = cv2.cvtColor(diff, cv2.COLOR_BGR2GRAY)
+    for i in range(0, 3):
+        dilated = cv2.dilate(gray.copy(), None, iterations=i + 1)
+    (t_var, thresh) = cv2.threshold(dilated, 3, 255, cv2.THRESH_BINARY)
+    cnts = cv2.findContours(thresh, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE)
+    cnts = imutils.grab_contours(cnts)
+    for contour in cnts:
+        (x, y, width, height) = cv2.boundingRect(contour)
+        cv2.rectangle(new, (x, y), (x + width, y + height), (0, 255, 0), 2)
+    return new
+
+
+def compare_images(browserdriver, base_image, test_image_name):
+    """Comparison logic and diff files creation."""
+    browserdriver.save_screenshot(f"{test_image_name}_branch.png")
```

Comment:
Does this get saved to a temporary directory? If it's all temporary then no need to cleanup with `rm` files afterwards.

Reply:
Should we just save these files to a temporary directory in the first place? e.g. in my suggested rewrite it could become:
```
def make_screenshot_paths(test_name, tmpdir):
    # result_image_path is something like tempdir / test_name.png
```

Then we can still do the `shutil.copy` as you do now and just no need to have the `Path().unlink()` line?

Reply:
It doesn't matter where the screenshot will be stored. I need to copy screenshots to artefacts folder for failed tests after all tests will be finished. Because I don't want to copy all of the screenshots, but only the ones related to failed tests, I'm deleting the screenshots for passed one right after comparison is done in a positive way. 

The other way can be to copy screenshots for failed tests into separate folder directly after tests is failed, but it will be more lines of code for copying and creating of the new folder (but the line for deleting files after passed tests could be deleted in this case)